quinn-proto/src/connection/mod.rs

```diff
@@ -2074,7 +2087,11 @@ where
                 if self.total_authed_packets > 1 {
                     return Ok(());
                 }
-                if packet.payload.chunks(4).any(is_supported_version) {
+                let supported = packet
+                    .payload
+                    .chunks(4)
+                    .any(|x| self.version == u32::from_be_bytes(x.try_into().unwrap()));
```

Comment:
Let's try avoiding the unwrap here?

```suggestion
                    .any(|x| match [u32; 4]::try_from(x) {
                        Ok(version) => self.version == version,
                        Err(_) => false,
                    })
```

Reply:
Or maybe `x.try_into().map_or(false, u32::from_be_bytes)`?

Reply:
I'm fine with that, too, though it seems a little harder to read to me.

Reply:
Ah, yeah, I guess the `match` is a bit less obscure even once you add the `u32::from_be_bytes`.